### ISSUE TITLE
Markup is being stripped in the export

### ISSUE THREAD

**Issue by RobErskine** [la-haute-societe/craft-page-exporter#2]
Sorry to open up a second issue 😅, we're getting super close.

We've noticed that some markup, most notably the doctype and opening/closing <html> tags are being stripped from the export.

On the left is the original source straight from our craft template, and the right is the exported .html file from Page Exporter.
![original code and exported code](https://cl.ly/8183abf63fb3/Screen%20Shot%202019-08-06%20at%2011.10.53%20AM.jpg)

The doctype:
```html
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
```

The opening html tag:

```html
<html xmlns="http://www.w3.org/1999/xhtml" xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office" lang="en-US">
```

And lastly the closing html tag:

```html
</html>
```

Is there a way to prevent these tags from being stripped? I also just want to note that we haven't purchased a license yet, but have every intention to do so. If it'd help to see that come through sooner than later, I'd be happy to pick it up.

Thanks for the time,
Rob

**Comment by RobErskine** [comment]
I think it has something to do with [Symfony's DOM](https://github.com/symfony/dom-crawler) crawler extension maybe? I noticed that the version used here is 3.1, while DOM crawler is up to v5. I tried updating the package to the latest and unfortunately, it didn't resolve this issue. 

I've added a very temporary solution that will get us to launch by replacing the addPage function in models/Export.php:
```php
public function addPage($pageName, $pageUrl, $content, $entry)
    {

        $updatedContent = '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml" xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office" lang="en-US">' . $content . '</html>';

        $currentPageAsset = new HtmlAsset([
            'name'            => $pageName . '.html',
            'url'             => $pageUrl,
            'fromString'      => $updatedContent, /** replace $content with new $updatedContent**/
            'export'          => $this,
            'baseUrl'         => $this->baseUrl,
            'basePath'        => $this->baseUrl,
            'customSelectors' => $this->customSelectors,
        ]);

        $currentPageAsset->populateChildren();
        $this->addRootAsset($currentPageAsset);

        return $this;
    }
```

I would love to help fix this in the long term though. If there's anything else I can do to troubleshoot please let me know.

**Comment by maxime-colin** [comment]
I pushed a fix in the release [1.1.5](https://github.com/la-haute-societe/craft-page-exporter/releases/tag/1.1.5), `html` and `doctype` tags should no longer be stripped. Let me know if it works for you.

PS: If you have to modify html content before exporting, you can use "[assetTransformers](https://github.com/la-haute-societe/craft-page-exporter#assettransformers)" in the configuration.

**Comment by RobErskine** [comment]
Thanks @maxime-colin, this looks good. 

And ah, that's good to know. I skipped over assetTransformers because I assumed it had something to do with craft's native asset transforms. Thanks again!
